src/replication/replication_server.rs

```diff
@@ -37,14 +75,45 @@ impl proto::replication_service_server::ReplicationService for ReplicationServer
         let request = request.into_inner();
 
         let snapshots = match self.replicator.get_snapshot_metadata(request.shard_id) {
-            Ok(metadata) => metadata
-                .into_iter()
-                .map(|(height, timestamp)| proto::ShardSnapshotMetadata {
-                    shard_id: request.shard_id,
-                    height,
-                    timestamp,
-                })
-                .collect(),
+            Ok(metadata) => {
+                let mut snapshots = Vec::new();
+                for (height, timestamp) in metadata {
+                    // Get the highest FID for this shard at this height
+                    let highest_fid = self
+                        .replicator
+                        .get_highest_fid_for_shard(request.shard_id, height)
+                        .unwrap_or(0);
+
+                    // Fetch the block for the given height
+                    let block = self.block_store.get_block_by_height(height).map_err(|e| {
+                        Status::internal(format!("Failed to get block by height: {}", e))
+                    })?;
```

Comment:
Why do we need the block?

Reply:
I thought we should insert the block into shard 0 as well, so that all shards are at the same block height and can start syncing. Let me know if you think this is not a good idea, and we should leave shard 0 empty

Reply:
Convo summary: We'll get the latest height for each shard separately.